File: 695/main.py
class Solution:
    def DFS(self, grid: List[List[int]], i: int, j: int) -> int:
        if i < 0 or j < 0 or i > len(grid) - 1 or j > len(grid[0]) - 1 or grid[i][j] == 0:
            return 0
        
        grid[i][j] = 0
        
        return 1 + \
    self.DFS(grid, i - 1, j) + \
    self.DFS(grid, i + 1, j) + \
    self.DFS(grid, i, j - 1) + \
    self.DFS(grid, i, j + 1)
    
    def maxAreaOfIsland(self, grid: List[List[int]]) -> int:
        maxArea = 0
        
        for i in range(len(grid)):
            for j in range(len(grid[0])):
                maxArea = max(maxArea, self.DFS(grid, i, j))
                
        return maxArea
        

File: 695/main.java
class Solution {
    public int maxAreaOfIsland(int[][] grid) {
        int maxArea = 0;
        
        for(int i = 0; i < grid.length; i++) {
            for(int j = 0; j < grid[0].length; j++) {
                maxArea = Math.max(maxArea, DFS(grid, i, j));
            }
        }
        
        return maxArea;
    }
    
    public int DFS(int[][] grid, int i, int j) {
        if(i < 0 || j < 0 || i > grid.length - 1 || j > grid[0].length - 1 || grid[i][j] == 0) {
            return 0;
        }
        
        grid[i][j] = 0;
        
        return 1 + 
            DFS(grid, i - 1, j) +
            DFS(grid, i + 1, j) + 
            DFS(grid, i, j - 1) +
            DFS(grid, i, j + 1);
    }
}
